editors/code/src/client.ts

```diff
@@ -65,22 +72,19 @@ export async function createClient(config: Config, serverPath: string): Promise<
         clientOptions,
     );
 
+    /* eslint-disable @typescript-eslint/ban-ts-ignore */
     // HACK: This is an awful way of filtering out the decorations notifications
     // However, pending proper support, this is the most effecitve approach
     // Proper support for this would entail a change to vscode-languageclient to allow not notifying on certain messages
     // Or the ability to disable the serverside component of highlighting (but this means that to do tracing we need to disable hihlighting)
     // This also requires considering our settings strategy, which is work which needs doing
     // @ts-ignore The tracer is private to vscode-languageclient, but we need access to it to not log publishDecorations requests
     res._tracer = {
```

Comment:
One way to avoid ts-ignore here at all is
```suggestion
    res['_tracer'] = {
```
but ts-ignore kinda brings your attention to the fact that this is a hack.